drivers/include/cc110x.h : Tagged compile config

akshaim opened the pull request:
### Contribution description

Identify Compile Time Parameters in drivers/include/cc110x.h header and expose it,

### Testing procedure

Doxygen build works fine.

### Issues/PRs references
~~Depends on #13784~~
Part of #10566

leandrolanzieri commented on drivers/include/cc110x.h:
```diff
@@ -240,13 +240,19 @@ extern "C" {
 
-#ifndef CC110X_DEFAULT_CHANNEL
+/**
+ * @defgroup drivers_cc110x_config     TI CC1100, CC110e and CC1101 driver compile configuration
```
```suggestion
 * @defgroup drivers_cc110x_config CC1100/CC1100e/CC1101 Sub-GHz transceiver driver compile configuration
```

leandrolanzieri commented:
ping @akshaim

akshaim commented:
@leandrolanzieri I have added the configurations to new subgroup and squashed. Please see if it okay , if yes I shall do a rebase to master. 

Also kindly merge the PR only after merging #13784

akshaim commented:
@leandrolanzieri  Please see if this can be merged now

leandrolanzieri commented:
@akshaim please rebase to current master

akshaim commented:
> @akshaim please rebase to current master

Done
